File: json/json.go
package main

import (
	"encoding/json"
	"fmt"
	"os"
	"strings"
)

func main() {
	dir := "companyfacts"
	files, err := os.ReadDir(dir)
	if err != nil {
		fmt.Fprintf(os.Stderr, "json: can't open %s: %v\n", dir, err)
		os.Exit(1)
	}

	for _, file := range files {
		if !strings.HasSuffix(file.Name(), ".json") {
			continue
		}

		name := dir + "/" + file.Name()
		jsonFile, err := os.Open(name) // For read access.
		if err != nil {
			fmt.Fprintf(os.Stderr, "json: can't open %s: %v\n", name, err)
			continue
		}
		defer jsonFile.Close()

		stat, err := jsonFile.Stat()
		if err != nil {
			fmt.Fprintf(os.Stderr, "json: can't stat %s: %v\n", name, err)
			continue
		}

		jsonBlob := make([]byte, stat.Size())
		_, err = jsonFile.Read(jsonBlob)
		if err != nil {
			fmt.Fprintf(os.Stderr, "json: can't read %s: %v\n", name, err)
			continue
		}

		var f interface{}
		err = json.Unmarshal(jsonBlob, &f)
		if err != nil {
			fmt.Fprintf(os.Stderr, "json: can't parse %s: %v\n", name, err)
			continue
		}
	}
}

File: json/json.php
<?php

$dir = "companyfacts";

if (!is_dir($dir)) {
    fprintf(STDERR, "json: can't open %s: directory not found\n", $dir);
    exit(1);
}

$files = scandir($dir);
if ($files === false) {
    fprintf(STDERR, "json: can't read directory %s\n", $dir);
    exit(1);
}

foreach ($files as $file) {
    if (!str_ends_with($file, ".json")) {
        continue;
    }
    
    $filepath = $dir . "/" . $file;
    
    $jsonContent = file_get_contents($filepath);
    if ($jsonContent === false) {
        fprintf(STDERR, "json: can't read %s\n", $filepath);
        continue;
    }
    
    $decoded = json_decode($jsonContent);
    if ($decoded === null && json_last_error() !== JSON_ERROR_NONE) {
        fprintf(STDERR, "json: can't parse %s: %s\n", $filepath, json_last_error_msg());
        continue;
    }
}

?>
